src/main/java/org/traccar/protocol/HuabaoProtocolDecoder.java

```diff
@@ -224,7 +225,7 @@ protected Object decode(
                         formatMessage(MSG_TERMINAL_REGISTER_RESPONSE, id, false, response), remoteAddress));
             }
 
-        } else if (type == MSG_TERMINAL_AUTH || type == MSG_HEARTBEAT || type == MSG_PHOTO) {
+        } else if (type == MSG_TERMINAL_AUTH || type == MSG_HEARTBEAT || type == MSG_PHOTO || type == MSG_HEARTBEAT_2) {
```

Comment:
```suggestion
        } else if (type == MSG_TERMINAL_AUTH || type == MSG_HEARTBEAT || type == MSG_HEARTBEAT_2 || type == MSG_PHOTO) {
```

Reply:
I can't see line length on Github editor - is the line too long or is it fine?

Reply:
I just swapped the order so that two heartbeats are next to each other.